File: session-05-chaos-engineering/python/order/service.py
from typing import Optional

from order.model import Order
from order.repository import OrderRepository
from kitchen.service import KitchenService, Preparation


class OrderService:
    def __init__(self, repo: OrderRepository, kitchen: KitchenService):
        self.repo = repo
        self.kitchen = kitchen

    def list_orders(self) -> list[Order]:
        return self.repo.find_all()

    def find_order(self, order_id: int) -> Optional[Order]:
        return self.repo.find_by_id(order_id)

    def place_order(self, order_id: int, drink: str) -> Preparation:
        return self.kitchen.prepare(order_id, drink)

File: session-05-chaos-engineering/go/order/service.go
package order

import "coffeeshop-chaos/kitchen"

type OrderService struct {
	repo    OrderRepository
	kitchen kitchen.KitchenService
}

func NewOrderService(repo OrderRepository, kitchen kitchen.KitchenService) *OrderService {
	return &OrderService{repo: repo, kitchen: kitchen}
}

func (s *OrderService) ListOrders() []Order {
	return s.repo.FindAll()
}

func (s *OrderService) FindOrder(id int) *Order {
	return s.repo.FindByID(id)
}

func (s *OrderService) PlaceOrder(orderID int, drink string) (*kitchen.Preparation, error) {
	return s.kitchen.Prepare(orderID, drink)
}
